sdk/core/azure-core/tests/perf_tests/_test_base.py

```diff
@@ -0,0 +1,354 @@
+        self.sync_transport = RequestsTransport
+        self.async_transport = AioHttpTransport
+
+        # if transport is specified, use that
+        if self.args.transport:
+            # if sync, override sync default
+            if self.args.sync:
+                try:
+                    self.sync_transport = sync_transport_types[self.args.transport]
+                except KeyError:
+                    raise ValueError(f"Invalid sync transport:{self.args.transport}\n Valid options are:\n- requests\n")
+            # if async, override async default
+            else:
+                try:
+                    self.async_transport = async_transport_types[self.args.transport]
+                except KeyError:
+                    raise ValueError(f"Invalid async transport:{self.args.transport}\n Valid options are:\n- aiohttp\n- requests\n")
+
+        self.error_map = {
+            401: ClientAuthenticationError,
+            404: ResourceNotFoundError,
+            409: ResourceExistsError,
+            304: ResourceNotModifiedError,
+        }
+    
+    def _build_sync_pipeline_client(self, auth_policy):
+        sync_core_policies = {
+            "UserAgentPolicy": UserAgentPolicy,
+            "HeadersPolicy": HeadersPolicy,
+            "ProxyPolicy": ProxyPolicy,
+            "NetworkTraceLoggingPolicy": NetworkTraceLoggingPolicy,
+            "HttpLoggingPolicy": HttpLoggingPolicy,
+            "RetryPolicy": RetryPolicy, 
+            "CustomHookPolicy": CustomHookPolicy,
+            "RedirectPolicy": RedirectPolicy,
+            "ContentDecodePolicy": ContentDecodePolicy,
+            "DistributedTracingPolicy": DistributedTracingPolicy,
+            "RequestIdPolicy": RequestIdPolicy,
+            "SensitiveHeaderCleanupPolicy": SensitiveHeaderCleanupPolicy,
+        } # last 4 are not included in default generated policies
+        default_policies = [
+            UserAgentPolicy,
+            HeadersPolicy,
+            ProxyPolicy,
+            NetworkTraceLoggingPolicy,
+            HttpLoggingPolicy,
+            RetryPolicy, 
+            CustomHookPolicy,
+            RedirectPolicy
+        ]
+
+        if self.args.policies is None:
+            # if None, only auth policy is passed in
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=[auth_policy]
+            )
+        elif self.args.policies == "all":
+            # if all, autorest default policies + auth policy
+            sync_policies = [policy(sdk_moniker=self.sdk_moniker) for policy in default_policies]
+            sync_policies.append(auth_policy)
+            sync_pipeline = Pipeline(
+                transport=self.sync_transport(),
+                policies=sync_policies
+            )
+        else:
+            # if custom list of policies, pass in custom list + auth policy
+            try:
+                custom_policies = ast.literal_eval(self.args.policies)
```

Comment:
Using `literal_eval` could possible accept broader input than we would like.
I'd probably go with something like:
```python
import azure.core.pipeline.policies as policies

try:
    custom_policies = [getattr(policies, p) for p in self.args.policies.split(',')
except AttributeError as e:
    raise ValueError(f"Azure Core has no policy named {e.name}. Please use policies from the following list...")

```

Reply:
updated